modules/video/src/libs/gifshot.js

```diff
@@ -1,2350 +0,0 @@
-                'options': options
-            });
-        }
-        else if (video) {
-            // If the user would like to create a GIF from an existing HTML5 video
-            existingVideo({
-                'existingVideo': video,
-                callback: callback,
-                options: options
-            });
-        }
-        else {
-            // If the user would like to create a GIF from a webcam stream
-            existingWebcam({
-                lastCameraStream: lastCameraStream,
-                callback: callback,
-                webcamVideoElement: webcamVideoElement,
-                options: options
-            });
-        }
-    }
-    /*
-      takeSnapShot.js
-      ===============
-    */
-    /* Copyright  2017 Yahoo Inc.
-     * Copyrights licensed under the MIT License. See the accompanying LICENSE file for terms.
-    */
-    function takeSnapShot(userOptions, callback) {
-        callback = utils.isFunction(userOptions) ? userOptions : callback;
-        userOptions = utils.isObject(userOptions) ? userOptions : {};
-        if (!utils.isFunction(callback)) {
-            return;
-        }
-        var mergedOptions = utils.normalizeOptions(defaultOptions, userOptions);
-        var options = utils.normalizeOptions(mergedOptions, {
-            'interval': .1,
-            'numFrames': 1,
-            'gifWidth': Math.floor(mergedOptions.gifWidth),
-            'gifHeight': Math.floor(mergedOptions.gifHeight)
-        });
-        createGIF(options, callback);
-    }
-    /*
-      API.js
-      ======
-    */
-    /* Copyright  2017 Yahoo Inc.
-     * Copyrights licensed under the MIT License. See the accompanying LICENSE file for terms.
-    */
-    // Dependencies
-    var API = {
-        'utils': utils$2,
-        'error': error$2,
-        'defaultOptions': defaultOptions$2,
-        'createGIF': createGIF,
-        'takeSnapShot': takeSnapShot,
-        'stopVideoStreaming': stopVideoStreaming,
-        'isSupported': isSupported,
-        'isWebCamGIFSupported': isWebCamGIFSupported,
-        'isExistingVideoGIFSupported': isExistingVideoGIFSupported,
-        'isExistingImagesGIFSupported': isSupported$1,
-        'VERSION': '0.4.5'
-    };
-    /*
-      index.js
-      ========
-    */
-    /* Copyright  2017 Yahoo Inc.
-     * Copyrights licensed under the MIT License. See the accompanying LICENSE file for terms.
-     */
-    // Universal Module Definition (UMD) to support AMD, CommonJS/Node.js, and plain browser loading
-    if (typeof define === 'function' && define.amd) {
-        define([], function () {
-            return API;
-        });
-    }
-    else if (typeof exports !== 'undefined') {
-        module.exports = API;
-    }
-    else {
-        window.gifshot = API;
-    }
-}(typeof window !== "undefined" ? window : {}, typeof document !== "undefined" ? document : { createElement: function () { } }, typeof window !== "undefined" ? window.navigator : {}));
```

Comment:
Removing this safe guard for `window` in the `.ts` rewrite introduced a bug in server side rendering when upgrading past version [v3.1.0-alpha.5](https://github.com/visgl/loaders.gl/releases/tag/v3.1.0-alpha.5), since the module is accessing window and it is not defined.

Last working stable version appears to be [3.0.12](https://github.com/visgl/loaders.gl/releases/tag/v3.0.12) 

@hkim8567 reported an error today from his next.js logs after a minor package upgrade: 
```
error - ReferenceError: window is not defined
ReferenceError: window is not defined
    at Object.<anonymous> (***/node_modules/@loaders.gl/video/dist/es5/lib/gifshot/gifshot.js:8:8)
```
This traces back to this code: `URL: window.URL || window.webkitURL || window.mozURL || window.msURL,`